### DISCUSSION TITLE
Binary (dfx) not running

### DISCUSSION THREAD

**Discussion by fgibbes** [ptitSeb/box64#152] [Q&A] [answered]
Hi,
I am trying to run dfx (the dfinity sdk for the Internet Computer) on a raspberry. DFX is a X86_64 binary. Here are the errors I get. Can you tell me what the issue is and whether there can be a solution?

![image](https://user-images.githubusercontent.com/91540043/138873812-53b703b3-a76f-4b76-ac73-207d08ae8aef.png)

**Comment by ptitSeb** [comment]
Those are function I need to wrap. I'll try to add them letter.
It seems `dfx` is launching at least, even without thos missing symbol

**Reply by fgibbes** [reply]
Thank you Seb for a fast response. I forgot to mention that dfx does not really start. It just displays the help information and this for any arguments I provide. 
It would be great if you could wrap those functions. Many people are looking for running dfx on a raspberry (only a x86_64 binary is available and compiling does not work for the moment). I would publish the success on the Internet Computer forum and mention Box64.
Gilbert

**Comment by ptitSeb** [comment]
is there a way to download this `dfx` app? I suspect it is using non-conventionnal way to get command line arguments. Or maybe it's a rust thing. I need to investigate.

**Reply by fgibbes** [reply]
It is a rust thing indeed. Here is a link where you can download the binary:  
https://sdk.dfinity.org/docs/developers-guide/install-upgrade-remove.html
Thanks again for your help. Merci.

**Comment by ptitSeb** [comment]
It's ok, I can reproduce the command line issue with a simple rust program? I don't need dfx for now.

**Comment by ptitSeb** [comment]
Ok, I think I have fixed the command line issue (and the missing symbols).
Can you try again?

**Comment by fgibbes** [comment]
Thanks Seb. I made a fast trial and I just got 1 error left:

![image](https://user-images.githubusercontent.com/91540043/138933443-844be424-58b6-4699-8a72-d3ae89e06d4c.png)

**Comment by ptitSeb** [comment, marked as answer]
I have added that last function, but I don't think it's related to error you see, the "File exists (os error 17)"
Not sure it's a box64 error here.

**Comment by fgibbes** [comment]
I tried again. You are right. It looks like this is not a box64 error. I guess with this I give up (for the moment).
Thanks a lot for your help.
